pkg/sync/sync_context.go

```diff
@@ -129,6 +129,14 @@ func WithPruneLast(enabled bool) SyncOpt {
 	}
 }
 
+// WithResourceModificationChecker sets resource modification checker
+func WithResourceModificationChecker(enabled bool, checker common.ModificationChecker) SyncOpt {
```

Comment:
Instead of asking consumer to re-comparing resources I would suggest passing `*diff.DiffResultList` : 

```
func WithApplyOutOfSyncOnly(enabled bool, diffResult *diff.DiffResultList) {
	if enabled {
		ctx.applyOutOfSyncDiffs = nil
	} else {
		ctx.applyOutOfSyncDiffs = diffResult
	}
}
```
The `*diff.DiffResultList` is always available prior to syncing.